docker/build_docker.sh

```diff
@@ -139,10 +139,35 @@ function docker_build {
         echo "== skipping image [${image_name}] as it is marked devonly =="
         return 0
     fi
+    
     VERSION=$(prop 'version' '1.0.0')    
     VERSION="${VERSION}.${REVISION}"
     echo "${image_name}: using version: ${VERSION}"
     image_full_name="${DOCKER_ORG}/${image_name}:${VERSION}"
+
+    if [[ "$(prop 'deprecated')" ]]; then
+        echo "${DOCKER_ORG}/${image_name} image is deprected, checking whethear the image is listed in the deprecated list or not"
+        reason=$(prop 'deprecated_reason')
+        
+        if ! ${PY3CMD} "${DOCKER_SRC_DIR}"/add_image_to_deprecated_or_internal_list.py "${DOCKER_ORG}"/"${image_name}" "${reason}" "${DOCKER_SRC_DIR}"/deprecated_images.json; then
+            echo "failed while handling deprected image"
+            return 1
+        else
+            echo "deprectaed no need to build"
+            return 0
+        fi
+    fi
+
+    if [[ "$(prop 'internalonly')" ]]; then
```

Comment:
Let's remove this part in the meantime, until we decide how to handle this case of internal docker image